src/main/java/io/openliberty/tools/common/plugins/config/ServerConfigDocument.java

```diff
@@ -238,6 +216,95 @@ private void initializeAppsLocation(CommonLoggerI log, File serverXML, File conf
+     * @throws Exception
+     */
+    public void processJvmOptions() throws FileNotFoundException, Exception {
+        final String jvmOptionsString = "jvm.options";
+        parsePropertiesFromFile(new File(libertyDirectoryPropertyToFile.get(ServerFeatureUtil.WLP_USER_DIR), jvmOptionsString));
+        parsePropertiesFromFile(getFileFromConfigDirectory(CONFIGDROPINS_DEFAULT + File.separator + jvmOptionsString));
+        parsePropertiesFromFile(getFileFromConfigDirectory(jvmOptionsString));
+        parsePropertiesFromFile(getFileFromConfigDirectory(CONFIGDROPINS_OVERRIDES + File.separator + jvmOptionsString));
+    }
+
+    /**
+     * server.env file read order
+     *   1. {wlp.install.dir}/etc/
+     *   2. {wlp.user.dir}/shared/
+     *   3. {server.config.dir}/
+     * @param serverEnvFile
+     * @throws Exception
+     * @throws FileNotFoundException
+     */
+    public void processServerEnv() throws Exception, FileNotFoundException {
+        final String serverEnvString = "server.env";
+        parsePropertiesFromFile(new File(libertyDirectoryPropertyToFile.get(ServerFeatureUtil.WLP_INSTALL_DIR), "etc" + File.separator + serverEnvString));
+        parsePropertiesFromFile(new File(libertyDirectoryPropertyToFile.get(ServerFeatureUtil.WLP_USER_DIR), "shared" + File.separator + serverEnvString));
+        parsePropertiesFromFile(new File(libertyDirectoryPropertyToFile.get(ServerFeatureUtil.SERVER_CONFIG_DIR), serverEnvString));
+    }
+
+    public void parsePropertiesFromFile(File propertiesFile) throws Exception, FileNotFoundException {
+        if (propertiesFile != null && propertiesFile.exists()) {
+            parseProperties(new FileInputStream(propertiesFile));
+        }
+    }
+
+    public void initializeFields(CommonLoggerI log, File serverXML, File configDir, Map<String, File> libertyDirPropertyFiles) {
+        this.log = log;
+        serverXMLFile = serverXML;
+        configDirectory = configDir;
+        if (libertyDirPropertyFiles != null) {
+            libertyDirectoryPropertyToFile = new HashMap<String, File>(libertyDirPropertyFiles);
+        } else {
+            log.warn("The properties for directories are null and could lead to application locations not being resolved correctly.");
+            libertyDirectoryPropertyToFile = new HashMap<String,File>();
+        }
+        locations = new HashSet<String>();
+        names = new HashSet<String>();
+        namelessLocations = new HashSet<String>();
+        locationsAndNames = new HashMap<String, String>();
+        props = new Properties();
+        defaultProps = new Properties();
+    }
+
+    /**
+     * Process bootstrap.properties and boostrap.include 
+     * @param bootstrapProp
+     * @param bootstrapFile - Optional specific file
+     * @throws Exception
+     * @throws FileNotFoundException
+     */
+    public void processBootstrapProperties(Map<String, String> bootstrapProp, File bootstrapFile) throws Exception, FileNotFoundException {
```

Comment:
Hopefully I am thinking correctly, but given what I was saying earlier I think we do not need to pass in separate props and files...just use whatever got copied/generated for bootstrap.properties in SERVER_CONFIG_DIR. Just need to make sure that `copyConfigFiles` is getting called before `ServerConfigDocument` is used in each case in LMP/LGP. I think it is.